src/entries/popup/pages/home/TokenDetails/TokenContextMenu.tsx

```diff
@@ -76,17 +92,55 @@ export function TokenContextMenu({ children, token }: TokenContextMenuProps) {
     navigate(ROUTES.BRIDGE);
   };
 
-  const isBridgeable = token.bridging?.isBridgeable;
+  const togglePin = useCallback(() => {
+    const toggle = () => {
+      if (pinned) {
+        removedPinnedAsset({ uniqueId: token.uniqueId });
+        return;
+      }
+      addPinnedAsset({ uniqueId: token.uniqueId });
+    };
+    toggle();
+    simulateClick(containerRef.current);
+  }, [
+    containerRef,
+    pinned,
+    addPinnedAsset,
+    token.uniqueId,
+    removedPinnedAsset,
+  ]);
 
-  const explorer = getTokenBlockExplorer(token);
-  const isNative = isNativeAsset(token?.address, token?.chainId);
+  const hideToken = useCallback(() => {
+    addHiddenAsset(token);
+    if (pinned) removedPinnedAsset({ uniqueId: token.uniqueId });
+    triggerToast({
+      title: i18n.t('token_details.more_options.hide_token_toast', {
+        name: token.symbol,
+      }),
+    });
+  }, [addHiddenAsset, removedPinnedAsset, pinned, token]);
+
+  const copyTokenAddress = useCallback(() => {
+    if (isNative) return;
+    navigator.clipboard.writeText(token.address);
+    triggerToast({
+      title: i18n.t('wallet_header.copy_toast'),
+      description: truncateAddress(token.address),
+    });
+    simulateClick(containerRef.current);
+  }, [isNative, token.address, containerRef]);
 
   if (isWatchingWallet && !allowSwap && isNative) return <>{children}</>;
 
   return (
     <DetailsMenuWrapper closed={true} onOpenChange={onOpenChange}>
       <ContextMenuTrigger asChild>{children}</ContextMenuTrigger>
       <ContextMenuContent>
+        <KeyboardShortcutListener
+          hideToken={hideToken}
+          togglePin={togglePin}
+          copyTokenAddress={copyTokenAddress}
+        />
```

Comment:
Have a look at how we do this for the tokens screen (home) with the `useTokensShortcuts` hook. I think we want to continue with consistency of that approach rather than a wrapper
https://github.com/rainbow-me/browser-extension/blob/764e5153b9c746cb7cf20f79ee46d6e103df3ec8/src/entries/popup/hooks/useTokensShortcuts.ts#L23